crates/analyzer/src/symbol_table.rs

```diff
@@ -452,22 +454,24 @@ impl SymbolTable {
         self.project_local_table.get(&prj).cloned()
     }
 
-    pub fn add_assign(
-        &mut self,
-        full_path: Vec<SymbolId>,
-        position: &AssignPosition,
-        partial: bool,
-    ) {
-        let assign = Assign {
-            path: AssignPath(full_path),
-            position: position.clone(),
-            partial,
-        };
-        self.assign_list.push(assign);
+    pub fn add_var_ref(&mut self, var_ref: &VarRef) {
+        let key = var_ref.affiniation;
+        self.var_ref_list.entry(key).or_default();
+
+        let list = self.var_ref_list.get_mut(&key).unwrap();
+        list.push(var_ref.clone());
```

Comment:
This three statements can be written like below:

```rust
self.var_ref_list
    .entry(key)
    .and_modify(|x| x.push(var_ref.clone()))
    .or_insert(vec![var_ref.clone()]);
```

By this way, fetching the entry becomes only once.